### PR TITLE
chore: react-virtuoso LogViewer companion

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
This is a companion PR to https://github.com/determined-ai/hew/pull/110.

## Test Plan
Check Trial Logs, Task Logs, and Cluster Logs to ensure they all work as expected.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by keita-determined** [comment]
iirc, we wanna resolve the double scrollbar issue, right (or is it out of scope)?
if so, i still see the issue
image

---

i see this bug in latest main, so not from this PR, but do you intend to fix this bug in this PR? The test plan scope isnt clear.

image


also, i saw some test failures.

**Comment by EmilyBonar** [comment]
Noticed that `TaskLogs.test.tsx` was really just testing LogViewer behavior, so I moved it over to hew. Due to the differences in library implementation some of the test cases had to be removed as well.

**Review by keita-determined — COMMENTED** [review]
Mostly good, but found the following bugs. Tested on Chrome

After pushing `Scroll the oldest`, i cant scroll down to the newest. 

https://github.com/determined-ai/determined/assets/109113616/6535856a-a624-4dba-b547-7329d8b127f1

---

There are double scrollbars with small viewport,.

https://github.com/determined-ai/determined/assets/109113616/6e7088ee-a5ea-4753-abe5-d56aa614263f


---

nit: In the full view, there is bottom margin. (see it in the latest-main website so can be ignored)

image

**Comment by EmilyBonar** [comment]
@keita-determined First bug should be fixed. The double scrollbar seems to be an issue with how the site as a whole handles resizing. In the last screenshot, I'm not sure what the issue is. The 16px margin is intentional and comes from Page.

**Review by keita-determined — COMMENTED** [review]
confirmed the first bug is fixed, and lets skip other 2 bugs as you said its not directly related to the pr.



idk why this happens, but it only happens on this branch. prob hew change caused this?

image

---

iirc, it was working for the commit, but now cannot search by text and/or dropdown filters

https://github.com/determined-ai/determined/assets/109113616/af2a728b-3472-45fc-815e-5510faa5dace

File: webui/react/package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "fp-ts": "^2.16.1",
     "fuse.js": "^7.0.0",
     "hermes-parallel-coordinates": "^0.6.17",
-    "hew": "npm:@hpe.com/hew@^0.6.31",
+    "hew": "github:determined-ai/hew.git#5eb7c907e0b69e9480c8dbc2d1a023a3483f74b0",
```
Comment (keita-determined) [inline webui/react/package.json:49]: nit: please make sure to use npm instead of commit hash before merging this PR

**Review by keita-determined — APPROVED** [review]
one more bug: search is not working by substring
in this example, search by `test` works, but not by `tes`

https://github.com/determined-ai/determined/assets/109113616/f8eaba0f-6354-4f3a-b202-faf9e35474ea

other than that, LGTM

**Comment by EmilyBonar** [comment]
@keita-determined That search bug seems like a blocker, but I'm not able to repro it. Did you do anything special/unusual?

**Comment by keita-determined** [comment]
> @keita-determined That search bug seems like a blocker, but I'm not able to repro it. Did you do anything special/unusual?

i first typed `test`, and delete a letter one by one slowly, then i saw the issue. I tested on Chrome.

**Comment by EmilyBonar** [comment]
@keita-determined Good catch, seem like there was a bug regarding resetting our list of IDs for dupe detection. Try it now and it should work.

**Comment by keita-determined** [comment]
yep, it works now

**Pull request merged**